packages/db/index.js

```diff
@@ -325,6 +325,98 @@ export class DBClient {
     }))
   }
 
+  /**
+   * List a users pinned files.
+   * @param {number} userId
+   * @param {import('./db-client-types').ListUploadsOptions} [opts]
+   * @returns {Promise<Array<import('./db-client-types').UploadItemOutput>>}
+   */
+  async listPinned (userId, opts = {}) {
+    // Query for pins the same shape as the uploads but with !inner joins
+    const pinnedQuery = `
+      _id:id::text,
+      name,
+      created:inserted_at,
+      updated:updated_at,
+      content!inner(
+        cid,
+        dagSize:dag_size,
+        pins:pin!inner(
+          status,
+          updated:updated_at,
+          location:pin_location(
+            _id:id,
+            peerId:peer_id,
+            peerName:peer_name,
+            ipfsPeerId:ipfs_peer_id,
+            region
+          )))`
+    const size = opts?.size || 10
+    // TODO get pinned for all of the users auth keys?
+    const authKeys = await this.listKeys(userId)
+    const authKeyIds = authKeys.map((authKey) => authKey._id)
+
+    let query = this._client
+      .from(psaPinRequestTableName)
+      .select(pinnedQuery)
+      .eq('content.pins.status', 'Pinned')
+      .in('auth_key_id', authKeyIds)
+      .is('deleted_at', null)
+      .limit(size)
+      .order(
+        opts.sortBy === 'Name' ? 'name' : 'inserted_at',
+        { ascending: opts.sortOrder === 'Asc' }
+      )
+
+    if (opts.before) {
+      query = query.lte('inserted_at', opts.before)
+    }
+
+    if (opts.after) {
+      query = query.gte('inserted_at', opts.after)
+    }
+
+    const { data: pins, error } = await query
+    if (error) {
+      throw new DBError(error)
+    }
+
+    const cids = pins?.map((u) => u.content.cid)
+    const deals = await this.getDealsForCids(cids)
+    return pins?.map((pin) => ({
+      ...normalizePsaPinRequest(pin),
+      type: 'PinRequest',
+      ...pin.content,
+      deals: deals[pin.content.cid] || []
+    }))
+  }
+
+  /**
+   * List a users files, both uploads and pinned.
+   * @param {number} userId
+   * @param {import('./db-client-types').ListUploadsOptions} [opts]
+   * @returns {Promise<Array<import('./db-client-types').UploadItemOutput>>}
+   */
+  async listUserFiles (userId, opts = {}) {
+    const uploadedUserFiles = await this.listUploads(userId, opts)
+    const pinnedUserFiles = await this.listPinned(userId, opts)
+
+    const userFiles = [
+      ...uploadedUserFiles,
+      ...pinnedUserFiles
+    ]
+
+    // Sort applied to combined array
+    if (opts.sortBy === 'Name') {
+      userFiles.sort((a, b) => ((opts.sortOrder === 'Asc') ? (a.name > b.name) : (a.name < b.name)) ? 1 : -1)
+    } else {
+      userFiles.sort((a, b) => ((opts.sortOrder === 'Asc') ? (new Date(a.created).getTime() - new Date(b.created).getTime()) : (new Date(b.created).getTime() - new Date(a.created).getTime())))
```

Comment:
I _think_ this could be simplified to:

```js
const sortProperty = options.sortBy === 'Name' ? 'name' : 'created'
userFiles.sort((a, b) => (((opts.sortOrder === 'Asc') === (a[sortProperty] > b[sortProperty])) ? 1 : -1)
```

Reply:
Yes, that's better. Have added test as well.

Reply:
NIce.